### PR TITLE
Add Guide for parallel processing in DataStore on iOS

### PR DESCRIPTION
**Pull request by Kilo-Loco**
*Issue #, if available:*
N/A

*Description of changes:*
Added a Guide that shows user how to perform batch CRUD operations using Combine for iOS. New section named "DataStore" was also added to the Guides.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by undefobj** [comment]
One suggestion - maybe we use a different phrase other than "batch" for this guide? [Batch terminology is something very specific with DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html) and I fear customers will get confused.

**Comment by drochetti** [comment]
> One suggestion - maybe we use a different phrase other than "batch" for this guide? [Batch terminology is something very specific with DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html) and I fear customers will get confused.

I second this. Even outside of the DynamoDB scenarios, batching has deeper implications, It think what it's happening in this PR is more about parallel processing than batching.

**Comment by aspittel** [comment]
I like the "in parallel" phrasing change and think this looks valuable, I'd love to hear more from some iOS focused folks on the team too.

File: docs/guides/datastore/fragments/parallel-processing-ios.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+To write data to the DataStore in parallel, pass an array of models to a Combine `Publishers.Sequence` and perform `Amplify.DataStore.save(_ model:)` on each object:
+
+```swift
+let todos = [
+    Todo(name: "Learn Amplify"),
+    Todo(name: "Build app"),
+    Todo(name: "Profit")
+]
+
+let sink = Publishers.Sequence<[Todo], Never>(sequence: todos)
```
Comment (palpatim) [inline docs/guides/datastore/fragments/parallel-processing-ios.md:20]: Here and below: can simplify this to `todos.publisher`

File: docs/guides/datastore/fragments/parallel-processing-ios.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    .sink {
+        if case let .failure(error) = $0 {
+            print("Error on save() for type Todo - \(error.localizedDescription)")
+        }
+    } receiveValue: { savedTodos in
+        print("Todos: \(savedTodos)")
+    }
+```
+
+`Publishers.Sequence` creates a `Publisher` where each model is passed to `.flatMap()` individually, allowing for each object to be saved to DataStore. `.collect()` emits a single array of all the elements collected, returning the saved models in `receiveValue`.
```
Comment (palpatim) [inline docs/guides/datastore/fragments/parallel-processing-ios.md:34]: "`todos.publisher` creates a ..."

**Review by palpatim — APPROVED** [review]
Very cool guide, and I also learned more about Combine. :) Minor notes below, but LGTM

**Review by aspittel — APPROVED** [review]

**Pull request merged**